vector_search/utils.py

```diff
@@ -199,25 +206,121 @@ def _chunk_documents(encoder, texts, metadatas):
     return recursive_splitter.create_documents(texts=texts, metadatas=metadatas)
 
 
+def _embedding_context(document, resource_type):
+    """
+    Get the embedding context for a document
+    based on the resource_type (learning resource vs content file)
+    """
+    if resource_type != CONTENT_FILE_TYPE:
+        return (
+            f"{document.get('title')} "
+            f"{document.get('description')} {document.get('full_description')}"
+        )
+    else:
+        return document.get("content", "")
+
+
 def _process_resource_embeddings(serialized_resources):
     docs = []
     metadata = []
     ids = []
     encoder = dense_encoder()
     vector_name = encoder.model_short_name()
     for doc in serialized_resources:
+        if not should_generate_embeddings(doc, COURSE_TYPE):
+            update_payload(doc, resource_type=COURSE_TYPE)
+            continue
         vector_point_key = doc["readable_id"]
         metadata.append(doc)
         ids.append(vector_point_id(vector_point_key))
-        docs.append(
-            f"{doc.get('title')} {doc.get('description')} {doc.get('full_description')}"
-        )
+        docs.append(_embedding_context(doc, COURSE_TYPE))
     if len(docs) > 0:
         embeddings = encoder.embed_documents(docs)
         return points_generator(ids, metadata, embeddings, vector_name)
     return None
 
 
+def update_payload(
+    serialized_document,
+    resource_type,
+):
+    """
+    Update the payload of a document in Qdrant
+    based on the resource_type (learning resource vs content file)
+    """
+    client = qdrant_client()
+    if resource_type != CONTENT_FILE_TYPE:
```

Comment:
Does it make sense to have a seperate version of update_payload(), should_generate_embeddings(),  _embedding_context() etc for learning resources and content files instead of having all the if statements?you should be able to know which version of the function to call in the task